src/OutputNode.cpp

```diff
@@ -56,10 +56,24 @@ GLuint OutputNode::render(WeakModel model) {
 }
 
 Chain OutputNode::chain() {
-    // Don't need to lock as long as m_chain is immutable
+    QMutexLocker locker(&d()->m_stateLock);
     return d()->m_chain;
 }
 
+void OutputNode::resize(QSize size) {
```

Comment:
I think this method isn't compatible with STRBON because STRBON moves the chain into the worker thread?

Reply:
should be good now (added a new constructor to `Chain` to preserve thread)